src/map/rodex.c

```diff
@@ -0,0 +1,581 @@
+/// @param amount : How much to remove
+void rodex_remove_item(struct map_session_data *sd, int16 idx, int16 amount)
+{
+	int i;
+	struct item *it;
+	struct item_data *itd;
+
+	nullpo_retv(sd);
+	Assert_retv(idx >= 0 && idx < MAX_INVENTORY);
+
+	for (i = 0; i < RODEX_MAX_ITEM; ++i) {
+		if (sd->rodex.tmp.items[i].idx == idx)
+			break;
+	}
+
+	if (i == RODEX_MAX_ITEM) {
+		clif->rodex_remove_item_result(sd, idx, -1);
+		return;
+	}
+
+	it = &sd->rodex.tmp.items[i].item;
+
+	if (amount <= 0 || amount > it->amount) {
+		clif->rodex_remove_item_result(sd, idx, -1);
+		return;
+	}
+
+	itd = itemdb->search(it->nameid);
+
+	if (amount == it->amount) {
+		sd->rodex.tmp.weight -= itd->weight * amount;
+		sd->rodex.tmp.items_count--;
+		if (sd->rodex.tmp.items_count < 1) {
+			sd->rodex.tmp.type &= ~MAIL_TYPE_ITEM;
+		}
+		memset(&sd->rodex.tmp.items[i], 0x0, sizeof(sd->rodex.tmp.items[0]));
+		clif->rodex_remove_item_result(sd, idx, 0);
+		return;
+	}
+
+	it->amount -= amount;
+	sd->rodex.tmp.weight -= itd->weight * amount;
+	
+	clif->rodex_remove_item_result(sd, idx, it->amount);
+}
+
+/// Request if character with given name exists and returns information about him
+/// @param sd : The player who's requesting
+/// @param name : The name of the character to check
+/// @param base_level : Reference to return the character base level, if he exists
+/// @param char_id : Reference to return the character id, if he exists
+/// @param class : Reference to return the character class id, if he exists
+void rodex_check_player(struct map_session_data *sd, const char *name, int *base_level, int *char_id, short *class)
+{
+	intif->rodex_checkname(sd, name);
+}
+
+/// Sends a Mail to an character
+/// @param sd : The player who's sending
+/// @param receiver_name : The name of the character who's receiving the message
+/// @param body : Mail message
+/// @param title : Mail Title
+/// @param zeny : Amount of zeny attached
+/// Returns result code:
+/// 	RODEX_SEND_MAIL_SUCCESS = 0,
+///		RODEX_SEND_MAIL_FATAL_ERROR = 1,
+///		RODEX_SEND_MAIL_COUNT_ERROR = 2,
+///		RODEX_SEND_MAIL_ITEM_ERROR = 3,
+///		RODEX_SEND_MAIL_RECEIVER_ERROR = 4
+int rodex_send_mail(struct map_session_data *sd, const char *receiver_name, const char *body, const char *title, int64 zeny)
+{
+	int i;
+	int64 total_zeny;
+
+	nullpo_retr(RODEX_SEND_MAIL_FATAL_ERROR, sd);
+	nullpo_retr(RODEX_SEND_MAIL_FATAL_ERROR, receiver_name);
+	nullpo_retr(RODEX_SEND_MAIL_FATAL_ERROR, body);
+	nullpo_retr(RODEX_SEND_MAIL_FATAL_ERROR, title);
+
+	total_zeny = zeny + sd->rodex.tmp.items_count * ATTACHITEM_COST + (2 * zeny)/100;
+	
+	if (strcmp(receiver_name, sd->rodex.tmp.receiver_name) != 0) {
+		rodex->clean(sd, 1);
+		return RODEX_SEND_MAIL_RECEIVER_ERROR;
+	}
+
+	if (total_zeny > sd->status.zeny || total_zeny < 0) {
+		rodex->clean(sd, 1);
+		return RODEX_SEND_MAIL_FATAL_ERROR;
+	}
+
+	rodex_refresh_stamps(sd);
+
+	if (sd->sc.data[SC_DAILYSENDMAILCNT]->val2 >= DAILY_MAX_MAILS) {
```

Comment:
i think better check. because init happened in other function. and from time it can be changed